packages/nextjs/src/client/routing/pagesRouterRoutingInstrumentation.ts

```diff
@@ -115,43 +111,37 @@ const client = getClient();
  * transaction names.
  */
 export function pagesRouterInstrumentation(
-  startTransactionCb: StartTransactionCb,
-  startTransactionOnPageLoad: boolean = true,
-  startTransactionOnLocationChange: boolean = true,
+  shouldInstrumentPageload: boolean,
+  shouldInstrumentNavigation: boolean,
   startPageloadSpanCallback: StartSpanCb,
   startNavigationSpanCallback: StartSpanCb,
 ): void {
   const { route, params, sentryTrace, baggage } = extractNextDataTagInformation();
-  // eslint-disable-next-line deprecation/deprecation
-  const { traceparentData, dynamicSamplingContext, propagationContext } = tracingContextFromHeaders(
-    sentryTrace,
-    baggage,
-  );
+  const { traceId, dsc, parentSpanId, sampled } = propagationContextFromHeaders(sentryTrace, baggage);
 
-  getCurrentScope().setPropagationContext(propagationContext);
   prevLocationName = route || globalObject.location.pathname;
 
-  if (startTransactionOnPageLoad) {
+  if (shouldInstrumentPageload) {
     const source = route ? 'route' : 'url';
-    const transactionContext = {
+    startPageloadSpanCallback({
       name: prevLocationName,
       op: 'pageload',
       origin: 'auto.pageload.nextjs.pages_router_instrumentation',
       tags: DEFAULT_TAGS,
       // pageload should always start at timeOrigin (and needs to be in s, not ms)
       startTimestamp: browserPerformanceTimeOrigin ? browserPerformanceTimeOrigin / 1000 : undefined,
       ...(params && client && client.getOptions().sendDefaultPii && { data: params }),
-      ...traceparentData,
+      parentSpanId,
```

Comment:
this will not work in the future, we'll need to refactor this to use only non-deprecated options (but we can do that in a follow up too)